src/treeview_sandbox.ml

```diff
@@ -79,6 +79,134 @@ module Command = struct
+                | Some `Install ->
+                  let (_ : Ojs.t option Promise.t) =
+                    Vscode.Commands.executeCommand
+                      ~command:"workbench.extensions.search"
+                      ~args:[ Ojs.string_to_js "ms-vscode.live-server" ]
+                  in
+                  ()
+              else
+                let arg = List.hd_exn args in
+                let dep = Dependency.t_of_js arg in
+                let name = Sandbox.Package.name dep in
+                let options =
+                  ProgressOptions.create
+                    ~location:(`ProgressLocation Notification)
+                    ~title:
+                      (Printf.sprintf "Generating documentation for %s" name)
+                    ~cancellable:false ()
+                in
+                let task ~progress:_ ~token:_ = Odig.odoc_exec odig name in
+                let* result =
+                  Vscode.Window.withProgress
+                    (module Interop.Js.Result
+                              (Interop.Js.String)
+                              (Interop.Js.String))
+                    ~options ~task
+                in
+                match result with
+                | Error _ ->
+                  let (_ : 'a option Promise.t) =
+                    Window.showErrorMessage
+                      ~message:
+                        (Printf.sprintf
+                           "Error while generating documentation for %s" name)
+                      ()
+                  in
+                  Promise.resolve ()
+                | Ok _ ->
+                  let+ html_dir = Odig.html_dir odig in
+                  let () =
+                    match sandbox with
+                    | Sandbox.Opam (_, Opam.Switch.Named _) ->
+                      let start =
+                        Vscode.Workspace.workspaceFolders () |> List.length
+                      in
+                      let workspaceFoldersToAdd :
+                          Vscode.Workspace.workspaceFolderToAdd =
+                        { name = "odig_doc"
+                        ; uri = Vscode.Uri.file (Path.to_string html_dir)
+                        }
+                      in
+                      let (_ : bool) =
+                        Vscode.Workspace.updateWorkspaceFolders ~start
+                          ~deleteCount:(Some 0)
+                          ~workspaceFoldersToAdd:[ workspaceFoldersToAdd ]
+                      in
+                      ()
+                    | Sandbox.Opam (_, Opam.Switch.Local _)
+                    | Sandbox.Esy (_, _)
+                    | Sandbox.Global
+                    | Sandbox.Custom _ ->
+                      ()
+                  in
+                  let htmlDocumentationPath = Path.(html_dir / name) in
+                  let (_ : Ojs.t option Promise.t) =
+                    Vscode.Commands.executeCommand
+                      ~command:"livePreview.start.preview.atFile"
```

Comment:
This is a builtin command, right? Then perhaps a typed binding for it should exist.

Reply:
After reading the other comments in this thread, I understand that this is not builtin. And neither is this thing very reliable. So never mind.